From e64ea63e0ddd0834024ddfce68c6ff82d75a5d91 Mon Sep 17 00:00:00 2001
From: Yi Zhao <yi.zhao@windriver.com>
Date: Fri, 3 Jul 2020 14:21:16 +0800
Subject: [PATCH] policy/modules/services/samba: allow smbd_t/nmbd_t to search
 tmpfs_t dir

Fixes:
avc:  denied  { search } for  pid=279 comm="nmbd" name="/" dev="tmpfs"
ino=12458 scontext=system_u:system_r:nmbd_t:s0-s15:c0.c1023
tcontext=system_u:object_r:tmpfs_t:s0 tclass=dir permissive=0

avc:  denied  { search } for  pid=344 comm="smbd" name="dbus"
dev="tmpfs" ino=14026 scontext=system_u:system_r:smbd_t:s0-s15:c0.c1023
tcontext=system_u:object_r:system_dbusd_runtime_t:s0 tclass=dir
permissive=0

Upstream-Status: Pending

Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 policy/modules/services/samba.te | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/policy/modules/services/samba.te b/policy/modules/services/samba.te
index 5287e5f0a..b084b5cc9 100644
--- a/policy/modules/services/samba.te
+++ b/policy/modules/services/samba.te
@@ -371,6 +371,7 @@ fs_search_auto_mountpoints(smbd_t)
 fs_getattr_rpc_dirs(smbd_t)
 fs_list_inotifyfs(smbd_t)
 fs_get_all_fs_quotas(smbd_t)
+fs_search_tmpfs(smbd_t)
 
 term_use_ptmx(smbd_t)
 
@@ -587,6 +588,7 @@ files_list_var_lib(nmbd_t)
 
 fs_getattr_all_fs(nmbd_t)
 fs_search_auto_mountpoints(nmbd_t)
+fs_search_tmpfs(nmbd_t)
 
 auth_use_nsswitch(nmbd_t)
 
-- 
2.17.1

